activerecord/lib/active_record/inheritance.rb

```diff
@@ -192,11 +192,11 @@ def type_condition(table = arel_table)
       # If this is a StrongParameters hash, and access to inheritance_column is not permitted,
```

Comment:
And to add a line to the CHANGELOG.md of ActiveRecord explaining that the behavior has changed: passing Parameters with an unpermitted `type` field will now raise an error (rather than being silently ignored)